src/lib/one-graph/cli-client.js

```diff
@@ -282,24 +283,68 @@ const updateGraphQLOperationsFileFromPersistedDoc = async (input) => {
   witnessedIncomingDocumentHashes.push(hash)
 }
 
-const handleCliSessionEvent = async ({ event, netlifyGraphConfig, netlifyToken, schema, siteId }) => {
+const handleCliSessionEvent = async ({ event, netlifyGraphConfig, netlifyToken, schema, sessionId, siteId }) => {
   const { __typename, payload } = await event
   switch (__typename) {
     case 'OneGraphNetlifyCliSessionTestEvent':
-      await handleCliSessionEvent({ netlifyToken, event: payload, netlifyGraphConfig, schema, siteId })
+      await handleCliSessionEvent({ netlifyToken, event: payload, netlifyGraphConfig, schema, sessionId, siteId })
       break
-    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent':
-      if (!payload.operationId || !payload.operationId.id) {
-        warn(`No operation id found in payload, ${JSON.stringify(payload, null, 2)}`)
+    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent': {
+      if (!payload.operationId && !payload.operation.id) {
+        warn(`No operation id found in payload,
+  ${JSON.stringify(payload, null, 2)}`)
         return
       }
-      generateHandlerByOperationId({
+      const files = generateHandlerByOperationId({
         netlifyGraphConfig,
         schema,
-        operationId: payload.operationId.id,
+        operationId: payload.operationId || payload.operation.id,
         handlerOptions: payload,
       })
+
+      if (!files) {
+        warn(`No files generated for operation id: ${payload.operationId || payload.operation.id}`)
+        return
+      }
+
+      const config = loadNetlifyGraphConfig()
+      for (const file of files) {
+        const fileWrittenEvent = {
+          __typename: 'OneGraphNetlifyCliSessionFileWrittenEvent',
+          cliSessionId: sessionId,
+          payload: {
+            editor: config.editor,
+            filepath: file.filePath,
+            audience: 'ui',
+          },
+        }
+
+        await OneGraphClient.executeCreateCLISessionEventMutation({
+          nfToken: netlifyToken,
+          sessionId,
+          payload: fileWrittenEvent,
+        })
+      }
+      break
+    }
+    case 'OneGraphNetlifyCliSessionOpenFileEvent': {
+      if (!payload.filePath) {
+        warn(`No filePath found in payload, ${JSON.stringify(payload, null, 2)}`)
+        return
+      }
+      const config = loadNetlifyGraphConfig()
+      if (config.editor) {
+        log(`Opening ${config.editor} for ${payload.filePath}`)
+        execa(config.editor, [payload.filePath], {
```

Comment:
Is it important to set some environment variables for the editor, or is it fine to inherit from the current process's env?

Reply:
I think we want to inherit the same env as the cli (which is presumably set up to work on the current site)